src/PolyType/Abstractions/AssociatedTypeAttributeAttribute.cs

```diff
@@ -14,7 +14,11 @@ public class AssociatedTypeAttributeAttribute : Attribute
     /// The name of the attribute constructor parameter or named argument (i.e. attribute property)
     /// whose <see cref="Type"/> or <see cref="Type"/> array argument should be included as associated types.
     /// </param>
-    /// <param name="requirements">The requirements to apply to each associated type.</param>
+    /// <param name="requirements">
+    /// The requirements to apply to each associated type.
+    /// Only applicable to object shapes.
+    /// All other shapes (e.g. collections, enums, unions) are always completely defined.
```

Comment:
I think this answers my earlier question. I could possibly nitpick the `None` vs. `Full` being applicable to all types but I can't see any real use-case right now. Union types are a bit nuanced since one might expect object hierarchies to inherit the annotation from the base, but I'm fine with leaving this underspecified for now.